src/content/docs/en/guides/migrate-to-astro/from-create-react-app.mdx

```diff
@@ -0,0 +1,379 @@
+yarn astro add react tailwind mdx
+```
+
+If your CRA requires any dependencies (e.g. NPM packages), then install them individually using the command line or by adding them to your new Astro project's `package.json` manually and then running an install command. Note that many, but not all, React dependencies will work in Astro. 
+
+### Add your existing app files
+
+Copy your existing Create React App (CRA) project source files and folders (e.g. `components`, `hooks`, `styles`, etc.) into a new folder inside `src/`, keeping its file structure so your app will continue to work. Note that all `.js` file extensions must be renamed to `.jsx` or `.tsx`.
+
+Do not include any configuration files. You will use Astro's own `astro.config.mjs`, `package.json`, and `tsconfig.json`.
+
+Move the contents of your app's `public/` folder (e.g. static assets) into Astro's `public/` folder.
+
+<FileTree>
+- public/
+  - logo.png
+  - favicon.ico
+  - ...
+- src/
+  - cra-project/
+    - App.jsx
+    - ...
+  - pages/
+    - index.astro
+- astro.config.mjs
+- package.json
+- tsconfig.json
+</FileTree>
+
+### Render your app
+
+Import your app's root component in the frontmatter section of `index.astro`, then render the `<App />` component in your page template:
+
+```astro title="src/pages/index.astro"
+---
+import App from '../cra-project/App.jsx'
+---
+<App client:load/>
+```
+
+:::note[client directives]
+Your app needs a [client directive](/en/reference/directives-reference/#client-directives) for interactivity. Astro will render your React app as static HTML until you opt-in to client-side JavaScript.
+
+Use `client:load` to ensure your app loads immediately from the server, or `client:only="react"` to skip server-rendering and run your app entirely client-side.
+:::
+
+### Check the Astro config file
+
+Astro has a configuration file at the root of your project called [`astro.config.mjs`](/en/guides/configuring-astro/). This is used for configuring your Astro project and any installed integrations, including [SSR adapters](/en/guides/deploy/).
+
+You should see the React integration installed, along with any other official Astro integrations you added.
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config'
+
+export default defineConfig({
+  integrations: {
+    [react()]
+  }
+  // your configuration options here...
+  // https://docs.astro.build/en/reference/configuration-reference/
+})
+```
+
+See our guide for [configuring Astro](/en/guides/configuring-astro/) for more details and available options.
+
+## Convert your Create React App to Astro
+
+After [adding your existing app to Astro](#add-your-create-react-app-to-astro), you will probably want to convert your app itself to Astro! 
+
+You will replicate a similar component-based design **using Astro HTML templating components for your basic structure** while importing and including individual React components (which may themselves be entire apps!) for islands of interactivity.
```

Comment:
I remember you seeing that Astro prefers linking between pages - is the possibility of a link where "using Astro HTML templating components for your basic structure" could be introduced?

I was thinking `/core-concepts/astro-components/` if it hasn't been previously introduced yet in this page.

Reply:
Yeah, I had thought about where exactly to link here (and had included it a tiny bit further down). It is a few paragraphs of plain text in a row, so visually it could totally take another link.  In fact, if we use the link, maybe we don't even need the bold? Maybe that is really what my subconscious was trying to do there!  :smile: Good call, let's try that link and see whether we can get away with no bold!